### ISSUE TITLE
`global_asm!` proc macro is not resolved

### ISSUE THREAD

**Issue by budde25** [rust-analyzer/rust-analyzer#9687]
[global_asm!](https://doc.rust-lang.org/beta/unstable-book/library-features/global-asm.html) proc macro throws there error that it is an [unresolved-macro-call](https://rust-analyzer.github.io/manual.html#unresolved-macro-call)
Error: "unresolved macro `self::global_asm!` rust-analyzer(unresolved-macro-call)" 

![image](https://user-images.githubusercontent.com/16231105/126871545-7c59c279-e9e2-402f-9a58-1a13857785cb.png)

this does compile output of cargo check:
```
cargo check
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
```

Also not sure if it is relevant but I also have `#![no_std]`

**Comment by jonas-schievink** [comment]
What Rust version are you using? This works fine for me. Can you provide the full source code needed to reproduce this?

**Comment by budde25** [comment]
rustc 1.55.0-nightly (67b03007c 2021-07-23)
rust-analyzer INFO [7/24/2021, 8:57:52 AM]: Extension version: 0.2.678

That is interesting because it has been constantly broken for me since I have been working on this project about 6 months ago. across different computers. I wonder if it has something to do with my project setup. The repo is https://github.com/budde25/os

**Comment by Andy-Python-Programmer** [comment]
I am experiencing the same bug.

## rustc version
`rustc 1.56.0-nightly (d9aa28767 2021-07-24)`

## rust analyzer version
`v0.2.678`

**Comment by Bobo1239** [comment]
Bisected the issue to https://github.com/rust-lang/rust/pull/87227 (included since `nightly-2021-07-20`) which probably makes sense. Don't have time to look into the rust-analyzer side but hope this helps.

**Comment by jonas-schievink** [comment]
Repro:

```rust
//- /lib.rs crate:lib deps:std
global_asm!();

//- /std.rs crate:std
pub mod prelude {
    pub mod rust_2018 {
        pub macro global_asm() {
            pub struct S;
        }
    }
}
```

This doesn't resolve, because we first check for `global_asm!` in the legacy scope, and then rewrite its path as `self::global_asm`. However, `self::global_asm` never resolves in the prelude.

This issue ties in with existing bugs around macro resolution described in https://github.com/rust-analyzer/rust-analyzer/issues/8828#issuecomment-872333960

**Comment by jonas-schievink** [comment]
fixed by #9700

**Comment by budde25** [comment]
Thank you!
